test/apps/network/tcp_poll_err.c

```diff
@@ -0,0 +1,263 @@
+// SPDX-License-Identifier: MPL-2.0
+
+#define _GNU_SOURCE
+
+#include <unistd.h>
+#include <sys/signal.h>
+#include <sys/socket.h>
+#include <sys/poll.h>
+#include <netinet/in.h>
+#include <arpa/inet.h>
+#include <fcntl.h>
+#include <stddef.h>
+
+#include "test.h"
+
+#define S_PORT htons(0x1238)
+
+static struct sockaddr_in sk_addr;
+int sk_listen;
+int sk_connect;
+int sk_accept;
+struct pollfd pfd = { .events = POLLIN | POLLOUT | POLLRDHUP };
+char buf[4096] = { 'a' };
+
+FN_TEST(poll_unconnected)
+{
+	sk_listen = CHECK(socket(PF_INET, SOCK_STREAM, 0));
+
+	pfd.fd = sk_listen;
+	TEST_RES(poll(&pfd, 1, 0), pfd.revents == (POLLOUT | POLLHUP));
+
+	sk_addr.sin_family = AF_INET;
+	sk_addr.sin_port = S_PORT;
+	CHECK(inet_aton("127.0.0.1", &sk_addr.sin_addr));
+	CHECK(bind(sk_listen, (struct sockaddr *)&sk_addr, sizeof(sk_addr)));
+
+	TEST_RES(poll(&pfd, 1, 0), pfd.revents == (POLLOUT | POLLHUP));
+}
+END_TEST()
+
+FN_TEST(poll_listen)
+{
+	CHECK(listen(sk_listen, 3));
+
+	TEST_RES(poll(&pfd, 1, 0), pfd.revents == 0);
+}
+END_TEST()
+
+FN_TEST(poll_connect_close)
+{
+	// connect -> close sk_connect -> accept sk_listen
```

Comment:
What does this comment mean? I don't understand how to interrupt the content, as it seems to be randomly mixed with system call names, variable names, and arrows.